Team417_CENTERSTAGE/src/main/java/org/firstinspires/ftc/team417_CENTERSTAGE/concepts/AprilTagDetectionConcept.java

```diff
@@ -0,0 +1,247 @@
+package org.firstinspires.ftc.team417_CENTERSTAGE.concepts;
+
+import android.util.Size;
+
+import com.acmerobotics.dashboard.FtcDashboard;
+import com.acmerobotics.dashboard.canvas.Canvas;
+import com.acmerobotics.dashboard.telemetry.TelemetryPacket;
+import com.acmerobotics.roadrunner.Pose2d;
+import com.qualcomm.robotcore.eventloop.opmode.Disabled;
+import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
+
+import org.firstinspires.ftc.robotcore.external.hardware.camera.BuiltinCameraDirection;
+import org.firstinspires.ftc.robotcore.external.hardware.camera.WebcamName;
+import org.firstinspires.ftc.team417_CENTERSTAGE.baseprograms.BaseTeleOp;
+import org.firstinspires.ftc.team417_CENTERSTAGE.apriltags.AprilTagInfoDump;
+import org.firstinspires.ftc.team417_CENTERSTAGE.utilityclasses.AprilTagInfo;
+import org.firstinspires.ftc.team417_CENTERSTAGE.utilityclasses.Pose;
+import org.firstinspires.ftc.team417_CENTERSTAGE.roadrunner.MecanumDrive;
+import org.firstinspires.ftc.vision.VisionPortal;
+import org.firstinspires.ftc.vision.apriltag.AprilTagDetection;
+import org.firstinspires.ftc.vision.apriltag.AprilTagProcessor;
+
+import java.util.ArrayList;
+
+/*
+    Concept: most of the code has gotten encapsulated in another class, but this does well for testing
```

Comment:
In that case it would probably be better to have this use the methods from the other class.